quinn/src/recv_stream.rs

```diff
@@ -486,6 +486,7 @@ where
         let this = self.get_mut();
         ready!(this.stream.poll_read(cx, &mut this.buf))?;
         match this.buf.filled().len() {
+            0 if this.buf.capacity() == 0 => Poll::Ready(Ok(Some(0))),
             0 => Poll::Ready(Ok(None)),
```

Comment:
Should we instead put the inverted condition on the existing 0 arm?

Reply:
Oh, duh.